devito/finite_differences/differentiable.py

```diff
@@ -277,19 +277,19 @@ def diff(self, *symbols, **assumptions):
         from devito.finite_differences.derivative import Derivative
         return Derivative(self, *symbols, **assumptions)
 
-    def _has(self, pattern):
+    def has(self, *pattern):
         """
-        Unlike generic SymPy use cases, in Devito the majority of calls to `_has`
+        Unlike generic SymPy use cases, in Devito the majority of calls to `has`
         occur through the finite difference routines passing `sympy.core.symbol.Symbol`
         as `pattern`. Since the generic `_has` can be prohibitively expensive,
-        we here quickly handle this special case, while using the superclass' `_has`
+        we here quickly handle this special case, while using the superclass' `has`
         as fallback.
         """
-        if isinstance(pattern, type) and issubclass(pattern, sympy.Symbol):
+        if isinstance(pattern[0], type) and issubclass(pattern[0], sympy.Symbol):
             # Symbols (and subclasses) are the leaves of an expression, and they
             # are promptly available via `free_symbols`. So this is super quick
-            return any(isinstance(i, pattern) for i in self.free_symbols)
-        return super(Differentiable, self)._has(pattern)
+            return any(isinstance(i, pattern[0]) for i in self.free_symbols)
+        return super(Differentiable, self).has(*pattern)
```

Comment:
I'm progressively ripping off all the unnecessary arguments to super. So this could be turned into:

```
return super().has(*pattern)
```

Reply:
sure